### ISSUE TITLE
`Select` does not allow to specify a generic argument

### ISSUE THREAD

**Issue by GerardasB** [iTwin/iTwinUI#1767]
#### Describe the bug (actual behavior)

Rendering a select with a generic argument `<Select<string> />` results in `Expected 0 type arguments, but got 1.ts(2558)` error.
Additionally, the implicit value type is `unknown`, resulting in `(_: unknown) => void | undefined` type for `onChange` prop.

#### Expected behavior
Allow specifying a generic argument for the `Select` component.

#### Additional information
[Version 3.2.1](https://www.npmjs.com/package/@itwin/itwinui-react/v/3.2.1)

```
// Select.d.ts
export declare const Select: React.ForwardRefExoticComponent<SelectProps<unknown> & React.RefAttributes<HTMLElement>>;
```

**Comment by mayank99** [comment]
Repro: https://stackblitz.com/edit/github-nubvvd?file=src%2FApp.tsx

This might have been caused by https://github.com/iTwin/iTwinUI/pull/1615. Need to investigate further.

**Comment by mayank99** [comment]
Yes, we need to fix the typescript issue while still forwarding the ref.

**Comment by mayank99** [comment]
fixed in 3.2.4
